--- thirdparty/Limine.h	2024-08-19 11:13:08.096677816 +1000
+++ Limine-modded.h	2024-08-19 11:22:05.846728681 +1000
@@ -293,6 +293,11 @@
 #define LIMINE_PAGING_MODE_MAX LIMINE_PAGING_MODE_LOONGARCH64_4LVL
 #define LIMINE_PAGING_MODE_MIN LIMINE_PAGING_MODE_LOONGARCH64_4LVL
 #define LIMINE_PAGING_MODE_DEFAULT LIMINE_PAGING_MODE_LOONGARCH64_4LVL
+#elif defined (__m68k__)
+#define LIMINE_PAGING_MODE_M68K_4K 0
+#define LIMINE_PAGING_MODE_M68K_8K 1
+#define LIMINE_PAGING_MODE_MAX LIMINE_PAGING_MODE_M68K_8K
+#define LIMINE_PAGING_MODE_DEFAULT LIMINE_PAGING_MODE_M68K_4K
 #else
 #error Unknown architecture
 #endif
@@ -404,6 +409,22 @@
     uint64_t cpu_count;
     LIMINE_PTR(struct limine_smp_info **) cpus;
 };
+
+#elif defined (__m68k__)
+
+struct limine_smp_info {
+    uint32_t processor_id;
+    uint64_t id;
+    LIMINE_PTR(limine_goto_address) goto_address;
+};
+
+struct limine_smp_response {
+    uint64_t revision;
+    uint32_t flags;
+    uint64_t bsp_id;
+    uint64_t cpu_count;
+    LIMINE_PTR(struct limine_smp_info **) cpus;
+};
 
 #else
 #error Unknown architecture
