diff -ru binutils-2.39/bfd/config.bfd binutils-2.39-sc/bfd/config.bfd
--- binutils-2.39/bfd/config.bfd	2022-07-08 10:46:47.000000000 +0100
+++ binutils-2.39-sc/bfd/config.bfd	2022-08-09 15:41:36.579017006 +0100
@@ -274,6 +274,11 @@
     targ_selvecs="aarch64_elf64_be_vec aarch64_elf32_le_vec aarch64_elf32_be_vec arm_elf32_le_vec arm_elf32_be_vec"
     want64=true
     ;;
+  aarch64-*-keyronex*)
+    targ_defvec=aarch64_elf64_le_vec
+    targ_selvecs=aarch64_elf32_le_vec
+    want64=true
+    ;;
   aarch64-*-linux* | aarch64-*-netbsd*)
     targ_defvec=aarch64_elf64_le_vec
     targ_selvecs="aarch64_elf64_be_vec aarch64_elf32_le_vec aarch64_elf32_be_vec arm_elf32_le_vec arm_elf32_be_vec aarch64_pei_vec"
@@ -720,6 +725,11 @@
     targ_selvecs=i386_elf32_vec
     want64=true
     ;;
+  x86_64-*-keyronex*)
+    targ_defvec=x86_64_elf64_vec
+    targ_selvecs=i386_elf32_vec
+    want64=true
+    ;;
 #endif
   i[3-7]86-*-lynxos*)
     targ_defvec=i386_elf32_vec
@@ -817,6 +827,9 @@
   m68*-*-haiku*)
     targ_defvec=m68k_elf32_vec
     ;;
+  m68*-*-keyronex*)
+    targ_defvec=m68k_elf32_vec
+    ;;
   m68hc11-*-* | m6811-*-*)
     targ_defvec=m68hc11_elf32_vec
     targ_selvecs="m68hc11_elf32_vec m68hc12_elf32_vec"
Only in binutils-2.39-sc/bfd: config.bfd.orig
diff -ru binutils-2.39/config.sub binutils-2.39-sc/config.sub
--- binutils-2.39/config.sub	2022-07-08 10:46:47.000000000 +0100
+++ binutils-2.39-sc/config.sub	2022-08-09 15:24:27.842536492 +0100
@@ -1754,7 +1754,7 @@
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* )
+	     | fiwix* | keyronex* )
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
diff -ru binutils-2.39/gas/configure.tgt binutils-2.39-sc/gas/configure.tgt
--- binutils-2.39/gas/configure.tgt	2022-07-08 10:46:47.000000000 +0100
+++ binutils-2.39-sc/gas/configure.tgt	2022-08-09 15:41:25.155767215 +0100
@@ -134,6 +134,7 @@
       aarch64*-linux-gnu_ilp32)		arch=aarch64:32 ;;
     esac ;;
   aarch64*-*-netbsd*)			fmt=elf em=nbsd;;
+  aarch64*-*-keyronex*) fmt=elf em=linux;;
 
   alpha-*-*vms*)			fmt=evax ;;
   alpha-*-osf*)				fmt=ecoff ;;
@@ -239,6 +240,7 @@
     esac ;;
   i386-*-lynxos*)			fmt=elf em=lynx ;;
   i386-*-redox*)			fmt=elf ;;
+  i386-*-keyronex*)			fmt=elf em=linux ;;
   i386-*-solaris*)			fmt=elf em=solaris ;;
   i386-*-freebsd* \
   | i386-*-kfreebsd*-gnu)		fmt=elf em=freebsd ;;
@@ -296,6 +298,8 @@
   m68k-*-gnu*)				fmt=elf ;;
   m68k-*-netbsd*)			fmt=elf em=nbsd ;;
   m68k-*-haiku*)			fmt=elf em=haiku ;;
+  m68k-*-keyronex*)			fmt=elf em=linux ;;
+
 
   s12z-*-*)				fmt=elf ;;
 
@@ -370,6 +374,7 @@
   pru-*-*)				fmt=elf ;;
 
   riscv*-*-haiku*)			fmt=elf endian=little em=haiku ;;
+  riscv*-*-haiku*)			fmt=elf endian=little em=linux ;;
   riscv*-*-*)				fmt=elf ;;
 
   rx-*-linux*)				fmt=elf em=linux ;;
diff -ru binutils-2.39/ld/configure.tgt binutils-2.39-sc/ld/configure.tgt
--- binutils-2.39/ld/configure.tgt	2022-07-29 08:37:48.000000000 +0100
+++ binutils-2.39-sc/ld/configure.tgt	2022-08-09 15:40:54.448095732 +0100
@@ -117,6 +117,9 @@
 aarch64-*-haiku*)	targ_emul=aarch64haiku
 			targ_extra_emuls="aarch64elf aarch64elf32 aarch64elf32b aarch64elfb armelf armelfb armelf_haiku $targ_extra_libpath"
 			;;
+aarch64-*-keyronex*)	targ_emul=aarch64elf
+			targ_extra_emuls="aarch64elfb armelf armelfb"
+			;;
 alpha*-*-freebsd* | alpha*-*-kfreebsd*-gnu)
 			targ_emul=elf64alpha_fbsd
 			targ_extra_emuls="elf64alpha alpha"
@@ -495,6 +498,8 @@
 			;;
 m68*-*-haiku*)		targ_emul=m68kelf
 			;;
+m68*-*-keyronex*)		targ_emul=m68kelfnbsd
+			;;
 m68*-*-*)		targ_emul=m68kelf
 			;;
 mcore-*-pe)		targ_emul=mcorepe ;
@@ -1007,6 +1012,9 @@
 x86_64-*-redox*)	targ_emul=elf_x86_64
 			targ_extra_emuls=elf_i386
 			;;
+x86_64-*-keyronex*)	targ_emul=elf_x86_64
+			targ_extra_emuls=elf_i386
+			;;
 x86_64-*-solaris2*)	targ_emul=elf_x86_64_sol2
 			targ_extra_emuls="elf_x86_64 elf_i386_sol2 elf_i386_ldso elf_i386 elf_iamcu"
 			targ_extra_libpath=$targ_extra_emuls
Only in binutils-2.39-sc/ld: configure.tgt.orig
